### PR TITLE
[pnpapi] Updates the PnP API

### PR DESCRIPTION
**Pull request by arcanis**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://next.yarnpkg.com/advanced/pnpapi
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

File: types/pnpapi/index.d.ts [hunk 1]
```diff
@@ -1,41 +1,60 @@
...
+    interface ProcessVersions {
+        pnp?: string;
+    }
 }
 
-export interface TopLevelPackageLocator {
-    name: null;
-    reference: null;
-}
+declare module 'pnpapi' {
```
Comment (RyanCavanaugh) [inline types/pnpapi/index.d.ts]: This is going to substantially worsen the end-user experience of this package. You can separate out the `NodeJS` augmentation into a different file and leave this file's format as-is.

**Review by RyanCavanaugh — CHANGES_REQUESTED** [review]

File: types/pnpapi/index.d.ts (hunk 1 above)
Comment (arcanis) [reply types/pnpapi/index.d.ts]: Sounds good, I'll check tomorrow. Why is the declare module approach discouraged btw? What use cases does it break?

**Comment by arcanis** [comment]
Ping @RyanCavanaugh ? 😃

File: types/pnpapi/tslint.json [hunk 2]
```diff
@@ -1 +1,8 @@
-{ "extends": "dtslint/dt.json" }
+{
+    "extends": "dtslint/dt.json",
+    "rules": {
+        "no-single-declare-module": false,
```
Comment (weswigham) [inline types/pnpapi/tslint.json:4]: In lieu of @RyanCavanaugh 's feedback - you should be able to remove these lint rule adjustments now, right?

**Review by weswigham — CHANGES_REQUESTED** [review]

**Pull request closed without merging**